api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto

```diff
@@ -70,4 +70,19 @@ message ProxyProtocol {
   //   :ref:`core.v3.ProxyProtocolConfig.pass_through_tlvs <envoy_v3_api_field_config.core.v3.ProxyProtocolConfig.pass_through_tlvs>`,
   //   which controls pass-through for the upstream.
   config.core.v3.ProxyProtocolPassThroughTLVs pass_through_tlvs = 3;
+
+  // The PROXY protocol versions to match. See https://www.haproxy.org/download/2.1/doc/proxy-protocol.txt for details.
+  // By default, the filter will match any version.
+  //
+  // When the filter receives PROXY protocol data that does not match the specified versions,
+  // it will reject the connection.
+  //
+  // .. attention::
+  //
+  //   When used in conjunction with the :ref:`allow_requests_without_proxy_protocol <envoy_v3_api_field_extensions.filters.listener.proxy_protocol.v3.ProxyProtocol.allow_requests_without_proxy_protocol>`,
+  //   the filter will only attempt to match signatures for the allowed versions.
+  //   For example, when ``allowed_versions=V1``, ``allow_requests_without_proxy_protocol=true``,
+  //   and an incoming request matches the V2 signature, the filter will allow the request through as if it did not contain
+  //   PROXY protocol information.
+  repeated config.core.v3.ProxyProtocolConfig.Version allowed_versions = 4;
```

Comment:
I'd like to get input from @envoyproxy/api-shepherds before you make any changes on this, but here's my thought:

* Would it make sense to have a list of disallowed versions instead of allowed versions? Then the empty-array is logically consistent behavior, instead of being a special case.
* Since there are only two versions, and it's unlikely there will ever be another version, should it just be two booleans `v1_disallowed` and `v2_disallowed` instead of a repeated field?

Reply:
Yeah, let's see what they say.

> Would it make sense to have a list of disallowed versions instead of allowed versions? Then the empty-array is logically consistent behavior, instead of being a special case.

Makes sense to me 👍 

> Since there are only two versions, and it's unlikely there will ever be another version, should it just be two booleans v1_disallowed and v2_disallowed instead of a repeated field?

This one I'm not too sure about. It would result in duplicate docs between the two fields unless we create a sub-message. Seems nicer to directly re-use the pre-existing PP version enum.

Reply:
I think I'd err towards disallowed.

Reply:
Done, thanks @htuch and @ggreenway for the suggestion.

Kept enum as is, I don't see much benefit for switching to bools.